libcob/intrinsic.c

```diff
@@ -1318,7 +1318,7 @@ cob_intr_day_of_integer (cob_field *srcdays)
 			leapyear = 365;
 		}
 	}
-	snprintf (buff, 15, "%4.4d%3.3d", baseyear, days);
+	snprintf (buff, 19, "%4.4d%3.3d", baseyear, days);
```

Comment:
Even if the second argument is 18, the warning messages will be solved.
Set the second argument to 18.